keps/sig-node/4265-proc-mount/README.md

```diff
@@ -0,0 +1,485 @@
+    // container runtimes mask certain paths in /proc to avoid accidental security
+    // exposure of special devices or information.
+    DefaultProcMount ProcMountType = "Default"
+
+    // UnmaskedProcMount bypasses the default masking behavior of the container
+    // runtime and ensures the newly created /proc the container stays in tact with
+    // no modifications.  
+    UnmaskedProcMount ProcMountType = "Unmasked"
+)
+
+procMount *ProcMountType
+```
+
+where nil is default, and is interpreted as "Default" ProcMountType.
+
+When the kubelet is presented with a pod that has a ProcMountType as Unmasked, it will edit the default list of
+masked paths it passes down to the [CRI to be empty](https://github.com/kubernetes/kubernetes/blob/964529b/pkg/securitycontext/util.go#L216) which it does
+with the [CRI request](https://github.com/kubernetes/kubernetes/blob/964529b/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L889-L891).
+
+This requires changes to the CRI runtime integrations so that kubelet will add the specific `unmasked` option.
+This was done after alpha:
+CRI-O has support in v1.25.0 after https://github.com/cri-o/cri-o/pull/6025/commits/4102586132214263c5d0ae93ec257432653ab82b
+containerd has support in 1.6. See https://github.com/containerd/containerd/pull/5070/commits/07f1df4541d6a81c205d194f4f6ea3a6a95c3e29
+
+The main use case for unmasking paths in `/proc` are for a user nesting unprivileged containers within a container. However, having an Unmasked ProcMountType
+is a privileged operation, and thus is part of the [privileged PSS policy](https://github.com/kubernetes/kubernetes/pull/103340). Since a user must have be in
+the privileged policy, they are also trusted to choose the correct user ID and run a workload that won't interfere with the host.
+
+A container running as root user on the host and an unmasked `/proc` could be able to write to the host `/proc`, and thus this privileged designation is appropriate.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a cluster admin, I would like a way to nest containers within containers. To do so, kernel the top level containers need an unmasked /proc.
+
+#### Story 2
+
+As a kubernetes user, I may want to build containers from within a kubernetes container. 
+See [this article for more information](https://github.com/jessfraz/blog/blob/master/content/post/building-container-images-securely-on-kubernetes.md).
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
```

Comment:
Risk: Placing more trust in the user namespacing mechanism of the kernel. (Arguably the kernel being responsible for making /proc and /sys secure is better anyway, just worth explicitly drawing attention to.)